assets/js/googlesitekit/datastore/modules/modules.js

```diff
@@ -0,0 +1,434 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+
+const { createRegistrySelector } = Data;
+
+// Actions
+const START_FETCH_SET_MODULE_STATUS = 'START_FETCH_SET_MODULE_STATUS';
+const FETCH_SET_MODULE_STATUS = 'FETCH_SET_MODULE_STATUS';
+const FINISH_FETCH_SET_MODULE_STATUS = 'FINISH_FETCH_SET_MODULE_STATUS';
+const CATCH_FETCH_SET_MODULE_STATUS = 'CATCH_FETCH_SET_MODULE_STATUS';
+const RECEIVE_SET_MODULE_STATUS = 'RECEIVE_SET_MODULE_STATUS';
+const START_FETCH_MODULES = 'START_FETCH_MODULES';
+const FETCH_MODULES = 'FETCH_MODULES';
+const FINISH_FETCH_MODULES = 'FINISH_FETCH_MODULES';
+const CATCH_FETCH_MODULES = 'CATCH_FETCH_MODULES';
+const RECEIVE_MODULES = 'RECEIVE_MODULES';
+
+export const INITIAL_STATE = {
+	modules: undefined,
+	isFetchingSetModuleStatus: {},
+	isFetchingModules: false,
+};
+
+export const actions = {
+	/**
+	 * Activate a module on the server.
+	 *
+	 * Activate a module (based on the slug provided).
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param  {string} slug Slug of the module to activate.
+	 * @return {Object}      Object with {response, error}
+	 */
+	*activateModule( slug ) {
+		const { response, error } = yield actions.setModuleStatus( slug, true );
+		return { response, error };
+	},
+
+	/**
+	 * Dectivate a module on the server.
+	 *
+	 * Dectivate a module (based on the slug provided).
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param  {string} slug Slug of the module to activate.
+	 * @return {Object}      Object with {response, error}
+	 */
+	*deactivateModule( slug ) {
+		const { response, error } = yield actions.setModuleStatus( slug, false );
+		return { response, error };
+	},
+
+	/**
+	 * (De)activate a module on the server.
+	 *
+	 * POSTs to the `core/modules/activation` endpoint to set the `active` status
+	 * supplied for the give `slug`.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param  {string}  slug   Slug of the module to activate/deactivate.
+	 * @param  {boolean} active `true` to activate; `false` to deactivate.
+	 * @return {Object}         Object with {response, error}
+	 */
+	*setModuleStatus( slug, active ) {
+		invariant( slug, 'slug is required.' );
+		invariant( active !== undefined, 'active is required.' );
+
+		let response, error;
+
+		yield {
+			payload: { active, slug },
```

Comment:
Not a big deal, but maybe we should turn these two into a `params` object right away, in preparation for #1288? This would not only be good to account for the meta-programming simplification, it's also generally a good idea to separate parameters passed from the response/error received (more relevant for e.g. `CATCH_FETCH_SET_MODULE_STATUS` below), like discussed the other week.